internal/xpkg/fuzz_test.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2023 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package xpkg
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/crossplane/crossplane-runtime/pkg/parser"
+
+	"github.com/spf13/afero"
+
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
+)
+
+func FuzzParse(f *testing.F) {
```

Comment:
This seems like it's testing code from crossplane-runtime, not crossplane. Similar to my comment above, I think this test belongs in crossplane-runtime.

Reply:
Totally agree, @AdamKorcz if you don't have strong reasons to keep these, I'd just remove both `xpkg.FuzzParse` and `revision.FuzzGCRExtract`

Reply:
On the other hand though, those are APIs we use and pass user provided inputs more or less directly into, and for example we can not leverage oss-fuzz for `crossplane-runtime`, nor we can push `go-containerregistry` to do fuzzing, so it might make sense to keep them

Reply:
IMO if the APIs are important to Crossplane, then it makes sense to keep them for the purpose of being aware if any issues exist in these APIs.

Reply:
AFAIK, the GCR parsing function is being fuzzed on OSS-Fuzz.

Reply:
I removed the test cases testing go-container registry and left the controller-runtime one, I'll have a look at whether we can lift FuzzParse to test `crossplane` logic instead of just `controller-runtime`.

Reply:
I didn't find an obvious improvement. I don't have strong opinions on this, let me know if you want me to remove also this test case, @negz.

Reply:
Yes, please remove this. I do want the test, but it belongs in crossplane-runtime, not here.